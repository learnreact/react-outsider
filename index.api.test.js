import React from "react"
import Outsider from "./"
import renderer from "react-test-renderer"

test("renders", () => {
  const subject = renderer.create(<Outsider />)
  expect(subject).toMatchSnapshot()
})

test("renders with children", () => {
  const subject = renderer.create(
    <Outsider>
      Hi, I'm <em>children</em>!
    </Outsider>
  )
  expect(subject).toMatchSnapshot()
})

test("renders with non-native api props removed", () => {
  const subject = renderer.create(
    <Outsider
      className="some-class"
      id="myId"
      style={null}
      onOutsideClick={() => {}}
      component="span"
    />
  )
  expect(subject).toMatchSnapshot()
})

test("renders as provided native 'component'", () => {
  const subject = renderer.create(<Outsider component="span" />)
  expect(subject.toJSON().type).toBe("span")
})

test("renders as provided custom 'component'", () => {
  const MyComponent = <section />
  const subject = renderer.create(<Outsider component="MyComponent" />)
  expect(subject.toJSON().type).toBe("MyComponent")
})
