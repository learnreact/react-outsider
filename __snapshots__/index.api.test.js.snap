// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `<div />`;

exports[`renders with children 1`] = `
<div>
  Hi, I'm 
  <em>
    children
  </em>
  !
</div>
`;

exports[`renders with non-native api props removed 1`] = `
<span
  className="some-class"
  id="myId"
  style={null}
/>
`;
